Adopt more smart pointers in WebCore/bindings

cdumez opened the pull request:
#### cce92d99566ed92ff5c4bd3ad1855f5f5cf88976
<pre>
Adopt more smart pointers in WebCore/bindings
<a href="https://bugs.webkit.org/show_bug.cgi?id=269366">https://bugs.webkit.org/show_bug.cgi?id=269366</a>

Reviewed by Brent Fulgham.

* Source/JavaScriptCore/runtime/JSObject.h:
* Source/WebCore/animation/WebAnimation.cpp:
(WebCore::WebAnimation::create):
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/bindings/js/CachedModuleScriptLoader.cpp:
(WebCore::CachedModuleScriptLoader::~CachedModuleScriptLoader):
(WebCore::CachedModuleScriptLoader::load):
(WebCore::CachedModuleScriptLoader::notifyFinished):
* Source/WebCore/bindings/js/CachedScriptSourceProvider.h:
(WebCore::CachedScriptSourceProvider::~CachedScriptSourceProvider):
(WebCore::CachedScriptSourceProvider::m_cachedScript):
(WebCore::CachedScriptSourceProvider::protectedScript const):
(WebCore::CachedScriptSourceProvider::hash const):
(WebCore::CachedScriptSourceProvider::source const):
* Source/WebCore/bindings/js/CommonVM.cpp:
(WebCore::commonVMSlow):
(WebCore::addImpureProperty):
(WebCore::protectedCommonVM):
* Source/WebCore/bindings/js/CommonVM.h:
* Source/WebCore/bindings/js/DOMGCOutputConstraint.h:
* Source/WebCore/bindings/js/DOMPromiseProxy.h:
(WebCore::DOMPromiseProxy&lt;IDLType&gt;::resolvePromise):
(WebCore::DOMPromiseProxy&lt;IDLUndefined&gt;::promise):
(WebCore::DOMPromiseProxyWithResolveCallback&lt;IDLType&gt;::promise):
* Source/WebCore/bindings/js/DOMWrapperWorld.cpp:
(WebCore::DOMWrapperWorld::~DOMWrapperWorld):
(WebCore::DOMWrapperWorld::clearWrappers):
(WebCore::DOMWrapperWorld::didCreateWindowProxy):
(WebCore::DOMWrapperWorld::didDestroyWindowProxy):
* Source/WebCore/bindings/js/DOMWrapperWorld.h:
(WebCore::DOMWrapperWorld::didCreateWindowProxy): Deleted.
(WebCore::DOMWrapperWorld::didDestroyWindowProxy): Deleted.
* Source/WebCore/bindings/js/GCController.cpp:
(WebCore::GCController::garbageCollectOnAlternateThreadForDebugging):
(WebCore::GCController::deleteAllCode):
(WebCore::GCController::deleteAllLinkedCode):
(WebCore::GCController::dumpHeap):
* Source/WebCore/bindings/js/IDBBindingUtilities.cpp:
(WebCore::get):
(WebCore::toJS):
(WebCore::createIDBKeyFromValue):
(WebCore::ensureNthValueOnKeyPath):
(WebCore::deserializeIDBValueToJSValue):
(WebCore::createKeyPathArray):
(WebCore::generateIndexKeyMapForValueIsolatedCopy):
(WebCore::deserializeIDBValueWithKeyInjection):
(WebCore::IDBSerializationContext::initializeVM):
* Source/WebCore/bindings/js/InternalReadableStream.cpp:
(WebCore::invokeReadableStreamFunction):
(WebCore::InternalReadableStream::isLocked const):
(WebCore::InternalReadableStream::isDisturbed const):
(WebCore::InternalReadableStream::cancel):
(WebCore::InternalReadableStream::lock):
(WebCore::InternalReadableStream::pipeTo):
(WebCore::InternalReadableStream::tee):
(WebCore::InternalReadableStream::getReader):
(WebCore::InternalReadableStream::pipeThrough):
* Source/WebCore/bindings/js/InternalWritableStream.cpp:
(WebCore::invokeWritableStreamFunction):
(WebCore::InternalWritableStream::createFromUnderlyingSink):
(WebCore::InternalWritableStream::locked const):
(WebCore::InternalWritableStream::lock):
(WebCore::InternalWritableStream::abortForBindings):
(WebCore::InternalWritableStream::closeForBindings):
(WebCore::InternalWritableStream::closeIfPossible):
(WebCore::InternalWritableStream::getWriter):
* Source/WebCore/bindings/js/JSCallbackData.cpp:
(WebCore::JSCallbackData::invokeCallback):
* Source/WebCore/bindings/js/JSCustomElementInterface.cpp:
(WebCore::JSCustomElementInterface::constructElementWithFallback):
(WebCore::JSCustomElementInterface::createElement):
(WebCore::JSCustomElementInterface::tryToConstructCustomElement):
(WebCore::constructCustomElementSynchronously):
(WebCore::JSCustomElementInterface::upgradeElement):
(WebCore::JSCustomElementInterface::invokeCallback):
(WebCore::JSCustomElementInterface::invokeFormStateRestoreCallback):
* Source/WebCore/bindings/js/JSCustomElementRegistryCustom.cpp:
(WebCore::getCustomElementCallback):
(WebCore::validateCustomElementNameAndThrowIfNeeded):
(WebCore::JSCustomElementRegistry::define):
(WebCore::whenDefinedPromise):
(WebCore::JSCustomElementRegistry::whenDefined):
* Source/WebCore/bindings/js/JSCustomEventCustom.cpp:
(WebCore::JSCustomEvent::detail const):
* Source/WebCore/bindings/js/JSDOMAbstractOperations.h:
(WebCore::isVisibleNamedProperty):
* Source/WebCore/bindings/js/JSDOMAsyncIterator.h:
(WebCore::IteratorTraits&gt;::runNextSteps):
(WebCore::IteratorTraits&gt;::getNextIterationResult):
(WebCore::IteratorTraits&gt;::onPromiseSettled):
(WebCore::IteratorTraits&gt;::createOnSettledFunction):
(WebCore::IteratorTraits&gt;::onPromiseFulFilled):
(WebCore::IteratorTraits&gt;::createOnFulfilledFunction):
(WebCore::IteratorTraits&gt;::reject):
(WebCore::IteratorTraits&gt;::onPromiseRejected):
(WebCore::IteratorTraits&gt;::createOnRejectedFunction):
(WebCore::IteratorTraits&gt;::next):
* Source/WebCore/bindings/js/JSDOMBindingSecurity.cpp:
(WebCore::printErrorMessageForFrame):
(WebCore::canAccessDocument):
(WebCore::BindingSecurity::shouldAllowAccessToFrame):
(WebCore::BindingSecurity::shouldAllowAccessToDOMWindow):
(WebCore::BindingSecurity::shouldAllowAccessToNode):
* Source/WebCore/bindings/js/JSDOMBindingSecurityInlines.h:
(WebCore::BindingSecurity::shouldAllowAccessToDOMWindow):
* Source/WebCore/bindings/js/JSDOMBuiltinConstructor.h:
(WebCore::JSDOMBuiltinConstructor&lt;JSClass&gt;::getDOMStructureForJSObject):
* Source/WebCore/bindings/js/JSDOMCastThisValue.h:
(WebCore::castThisValue):
* Source/WebCore/bindings/js/JSDOMConstructorBase.cpp:
(WebCore::JSC_DEFINE_HOST_FUNCTION):
* Source/WebCore/bindings/js/JSDOMConvertAny.h:
(WebCore::VariadicConverter&lt;IDLAny&gt;::convert):
* Source/WebCore/bindings/js/JSDOMConvertBufferSource.h:
(WebCore::Detail::BufferSourceConverter::convert):
* Source/WebCore/bindings/js/JSDOMConvertCallbacks.h:
(WebCore::Converter&lt;IDLCallbackFunction&lt;T&gt;&gt;::convert):
(WebCore::Converter&lt;IDLCallbackInterface&lt;T&gt;&gt;::convert):
* Source/WebCore/bindings/js/JSDOMConvertDate.cpp:
(WebCore::valueToDate):
* Source/WebCore/bindings/js/JSDOMConvertEnumeration.h:
(WebCore::Converter&lt;IDLEnumeration&lt;T&gt;&gt;::convert):
* Source/WebCore/bindings/js/JSDOMConvertEventListener.h:
(WebCore::Converter&lt;IDLEventListener&lt;T&gt;&gt;::convert):
* Source/WebCore/bindings/js/JSDOMExceptionHandling.cpp:
(WebCore::reportException):
(WebCore::reportCurrentException):
(WebCore::createDOMException):
* Source/WebCore/bindings/js/JSDOMGlobalObject.cpp:
(WebCore::JSC_DEFINE_HOST_FUNCTION):
(WebCore::JSDOMGlobalObject::protectedScriptExecutionContext const):
(WebCore::JSDOMGlobalObject::promiseRejectionTracker):
(WebCore::JSDOMGlobalObject::createCrossOriginFunction):
(WebCore::JSDOMGlobalObject::createCrossOriginGetterSetter):
(WebCore::handleResponseOnStreamingAction):
(WebCore::JSDOMGlobalObject::moduleLoaderFetch):
(WebCore::JSDOMGlobalObject::moduleLoaderImportModule):
(WebCore::JSDOMGlobalObject::deriveShadowRealmGlobalObject):
(WebCore::JSDOMGlobalObject::protectedWorld):
(WebCore::toJSDOMGlobalObject):
(WebCore::callerGlobalObject):
* Source/WebCore/bindings/js/JSDOMGlobalObject.h:
* Source/WebCore/bindings/js/JSDOMGuardedObject.cpp:
(WebCore::DOMGuardedObject::DOMGuardedObject):
* Source/WebCore/bindings/js/JSDOMIterator.cpp:
(WebCore::addValueIterableMethods):
* Source/WebCore/bindings/js/JSDOMIterator.h:
(WebCore::iteratorForEach):
(WebCore::IteratorTraits&gt;::next):
* Source/WebCore/bindings/js/JSDOMMapLike.h:
(WebCore::forwardSizeToMapLike):
(WebCore::forwardEntriesToMapLike):
(WebCore::forwardKeysToMapLike):
(WebCore::forwardValuesToMapLike):
(WebCore::forwardClearToMapLike):
(WebCore::forwardGetToMapLike):
(WebCore::forwardHasToMapLike):
(WebCore::forwardSetToMapLike):
(WebCore::forwardDeleteToMapLike):
* Source/WebCore/bindings/js/JSDOMMicrotask.cpp:
(WebCore::JSDOMMicrotask::run):
* Source/WebCore/bindings/js/JSDOMOperation.h:
(WebCore::IDLOperation::call):
* Source/WebCore/bindings/js/JSDOMPromise.cpp:
(WebCore::DOMPromise::whenPromiseIsSettled):
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.cpp:
(WebCore::DeferredPromise::callFunction):
(WebCore::DeferredPromise::whenSettled):
(WebCore::DeferredPromise::reject):
(WebCore::createRejectedPromiseWithTypeError):
(WebCore::DeferredPromise::handleTerminationExceptionIfNeeded):
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.h:
(WebCore::DeferredPromise::resolve):
(WebCore::DeferredPromise::resolveWithNewlyCreated):
(WebCore::DeferredPromise::resolveCallbackValueWithNewlyCreated):
(WebCore::DeferredPromise::reject):
(WebCore::DeferredPromise::resolveWithCallback):
(WebCore::DeferredPromise::rejectWithCallback):
(WebCore::DOMPromiseDeferredBase::reject):
(WebCore::DOMPromiseDeferredBase::rejectType):
(WebCore::DOMPromiseDeferredBase::whenSettled):
(WebCore::DOMPromiseDeferredBase::protectedPromise const):
(WebCore::DOMPromiseDeferred::resolve):
(WebCore::DOMPromiseDeferred&lt;void&gt;::resolve):
(WebCore::callPromiseFunction):
(WebCore::callPromisePairFunction):
* Source/WebCore/bindings/js/JSDOMSetLike.cpp:
(WebCore::getBackingSet):
(WebCore::clearBackingSet):
(WebCore::addToBackingSet):
(WebCore::forwardFunctionCallToBackingSet):
* Source/WebCore/bindings/js/JSDOMSetLike.h:
(WebCore::getAndInitializeBackingSet):
(WebCore::forwardSizeToSetLike):
(WebCore::forwardEntriesToSetLike):
(WebCore::forwardKeysToSetLike):
(WebCore::forwardValuesToSetLike):
(WebCore::forwardClearToSetLike):
(WebCore::forwardHasToSetLike):
(WebCore::forwardAddToSetLike):
(WebCore::forwardDeleteToSetLike):
* Source/WebCore/bindings/js/JSDOMWindowBase.cpp:
(WebCore::JSDOMWindowBase::updateDocument):
(WebCore::JSDOMWindowBase::printErrorMessage const):
(WebCore::JSDOMWindowBase::supportsRichSourceInfo):
(WebCore::JSDOMWindowBase::shouldInterruptScript):
(WebCore::JSDOMWindowBase::shouldInterruptScriptBeforeTimeout):
(WebCore::JSDOMWindowBase::javaScriptRuntimeFlags):
(WebCore::JSDOMWindowBase::queueMicrotaskToEventLoop):
(WebCore::JSDOMWindowBase::currentScriptExecutionOwner):
(WebCore::JSDOMWindowBase::fireFrameClearedWatchpointsForWindow):
* Source/WebCore/bindings/js/JSDOMWindowProperties.cpp:
(WebCore::jsDOMWindowPropertiesGetOwnPropertySlotNamedItemGetter):
(WebCore::JSDOMWindowProperties::finishCreation):
(WebCore::JSDOMWindowProperties::getOwnPropertySlot):
(WebCore::JSDOMWindowProperties::getOwnPropertySlotByIndex):
(WebCore::JSDOMWindowProperties::defineOwnProperty):
* Source/WebCore/bindings/js/JSDOMWrapper.cpp:
(WebCore::cloneAcrossWorlds):
* Source/WebCore/bindings/js/JSDOMWrapperCache.h:
(WebCore::setSubclassStructureIfNeeded):
* Source/WebCore/bindings/js/JSDocumentCustom.cpp:
(WebCore::cachedDocumentWrapper):
(WebCore::reportMemoryForDocumentIfFrameless):
(WebCore::setAdoptedStyleSheetsOnTreeScope):
(WebCore::JSDocument::setAdoptedStyleSheets):
* Source/WebCore/bindings/js/JSElementCustom.cpp:
(WebCore::toJS):
(WebCore::toJSNewlyCreated):
(WebCore::getElementsArrayAttribute):
* Source/WebCore/bindings/js/JSElementInternalsCustom.cpp:
(WebCore::JSElementInternals::setFormValue):
(WebCore::getElementsArrayAttribute):
* Source/WebCore/bindings/js/JSErrorHandler.cpp:
(WebCore::JSErrorHandler::handleEvent):
* Source/WebCore/bindings/js/JSEventListener.cpp:
(WebCore::JSEventListener::handleEvent):
(WebCore::JSEventListener::functionName const):
* Source/WebCore/bindings/js/JSEventListener.h:
(WebCore::windowEventHandlerAttribute):
(WebCore::setWindowEventHandlerAttribute):
(WebCore::JSEventListener::ensureJSFunction const):
* Source/WebCore/bindings/js/JSEventTargetCustom.h:
(WebCore::JSEventTargetWrapper::JSEventTargetWrapper):
(WebCore::IDLOperation&lt;JSEventTarget&gt;::call):
* Source/WebCore/bindings/js/JSExecState.cpp:
(WebCore::JSExecState::didLeaveScriptContext):
* Source/WebCore/bindings/js/JSExecState.h:
(WebCore::JSExecState::call):
(WebCore::JSExecState::evaluate):
(WebCore::JSExecState::profiledCall):
(WebCore::JSExecState::profiledEvaluate):
(WebCore::JSExecState::linkAndEvaluateModule):
(WebCore::JSExecState::~JSExecState):
* Source/WebCore/bindings/js/JSExtendableMessageEventCustom.cpp:
(WebCore::constructJSExtendableMessageEvent):
(WebCore::JSExtendableMessageEvent::data const):
* Source/WebCore/bindings/js/JSHTMLAllCollectionCustom.cpp:
(WebCore::JSC_DEFINE_HOST_FUNCTION):
* Source/WebCore/bindings/js/JSHTMLElementCustom.cpp:
(WebCore::constructJSHTMLElement):
(WebCore::JSHTMLElement::pushEventHandlerScope const):
* Source/WebCore/bindings/js/JSHistoryCustom.cpp:
(WebCore::JSHistory::state const):
* Source/WebCore/bindings/js/JSImageDataCustom.cpp:
(WebCore::toJSNewlyCreated):
* Source/WebCore/bindings/js/JSKeyframeEffectCustom.cpp:
(WebCore::JSKeyframeEffect::getKeyframes):
* Source/WebCore/bindings/js/JSLazyEventListener.cpp:
(WebCore::JSLazyEventListener::JSLazyEventListener):
(WebCore::JSLazyEventListener::initializeJSFunction const):
(WebCore::JSLazyEventListener::create):
* Source/WebCore/bindings/js/JSLocalDOMWindowCustom.cpp:
(WebCore::JSC_DEFINE_CUSTOM_GETTER):
(WebCore::jsLocalDOMWindowGetOwnPropertySlotRestrictedAccess):
(WebCore::JSLocalDOMWindow::getOwnPropertySlot):
(WebCore::JSLocalDOMWindow::getOwnPropertySlotByIndex):
(WebCore::JSLocalDOMWindow::put):
(WebCore::JSLocalDOMWindow::putByIndex):
(WebCore::JSLocalDOMWindow::deleteProperty):
(WebCore::JSLocalDOMWindow::deletePropertyByIndex):
(WebCore::addCrossOriginOwnPropertyNames):
(WebCore::addScopedChildrenIndexes):
(WebCore::JSLocalDOMWindow::getOwnPropertyNames):
(WebCore::JSLocalDOMWindow::defineOwnProperty):
(WebCore::JSLocalDOMWindow::getPrototype):
(WebCore::DialogHandler::dialogCreated):
(WebCore::DialogHandler::returnValue const):
(WebCore::JSC_DEFINE_HOST_FUNCTION):
(WebCore::JSLocalDOMWindow::queueMicrotask):
(WebCore::JSLocalDOMWindow::setOpener):
(WebCore::jsLocalDOMWindowInstanceFunction_openDatabaseBody):
(WebCore::JSLocalDOMWindow::openDatabase const):
(WebCore::JSLocalDOMWindow::setOpenDatabase):
* Source/WebCore/bindings/js/JSLocationCustom.cpp:
(WebCore::getOwnPropertySlotCommon):
(WebCore::JSLocation::getOwnPropertySlot):
(WebCore::JSLocation::getOwnPropertySlotByIndex):
(WebCore::JSLocation::put):
(WebCore::JSLocation::putByIndex):
(WebCore::JSLocation::deleteProperty):
(WebCore::JSLocation::deletePropertyByIndex):
(WebCore::JSLocation::getOwnPropertyNames):
(WebCore::JSLocation::defineOwnProperty):
(WebCore::JSLocation::getPrototype):
* Source/WebCore/bindings/js/JSMicrotaskCallback.h:
(WebCore::JSMicrotaskCallback::call):
* Source/WebCore/bindings/js/JSNodeCustom.cpp:
(WebCore::willCreatePossiblyOrphanedTreeByRemovalSlowCase):
* Source/WebCore/bindings/js/JSObservableArray.cpp:
(JSC::JSC_DEFINE_CUSTOM_GETTER):
(JSC::JSObservableArray::getOwnPropertyNames):
(JSC::JSObservableArray::getOwnPropertySlot):
(JSC::JSObservableArray::put):
(JSC::JSObservableArray::deleteProperty):
(JSC::JSObservableArray::defineOwnProperty):
* Source/WebCore/bindings/js/JSObservableArray.h:
* Source/WebCore/bindings/js/JSPopStateEventCustom.cpp:
(WebCore::JSPopStateEvent::state const):
* Source/WebCore/bindings/js/JSRTCRtpSFrameTransformCustom.cpp:
(WebCore::JSRTCRtpSFrameTransform::setEncryptionKey):
* Source/WebCore/bindings/js/JSReadableStreamSourceCustom.cpp:
(WebCore::JSReadableStreamSource::start):
(WebCore::JSReadableStreamSource::pull):
* Source/WebCore/bindings/js/JSRemoteDOMWindowBase.cpp:
(WebCore::JSRemoteDOMWindowBase::protectedWrapped const):
* Source/WebCore/bindings/js/JSRemoteDOMWindowBase.h:
* Source/WebCore/bindings/js/JSRemoteDOMWindowCustom.cpp:
(WebCore::JSRemoteDOMWindow::getOwnPropertySlotByIndex):
(WebCore::JSRemoteDOMWindow::put):
(WebCore::JSRemoteDOMWindow::deleteProperty):
(WebCore::JSRemoteDOMWindow::deletePropertyByIndex):
(WebCore::JSRemoteDOMWindow::defineOwnProperty):
* Source/WebCore/bindings/js/JSShadowRootCustom.cpp:
(WebCore::JSShadowRoot::setAdoptedStyleSheets):
* Source/WebCore/bindings/js/JSWebAnimationCustom.cpp:
(WebCore::constructJSWebAnimation):
* Source/WebCore/bindings/js/JSWindowProxy.cpp:
(WebCore::JSWindowProxy::setWindow):
* Source/WebCore/bindings/js/JSWorkerGlobalScopeBase.cpp:
(WebCore::JSWorkerGlobalScopeBase::queueMicrotaskToEventLoop):
* Source/WebCore/bindings/js/JSWorkerGlobalScopeCustom.cpp:
(WebCore::JSWorkerGlobalScope::queueMicrotask):
* Source/WebCore/bindings/js/ModuleScriptLoader.h:
(WebCore::ModuleScriptLoader::ModuleScriptLoader):
* Source/WebCore/bindings/js/ModuleScriptLoaderClient.h:
* Source/WebCore/bindings/js/ReadableStreamDefaultController.cpp:
(WebCore::invokeReadableStreamDefaultControllerFunction):
(WebCore::ReadableStreamDefaultController::error):
(WebCore::ReadableStreamDefaultController::enqueue):
* Source/WebCore/bindings/js/ScheduledAction.cpp:
(WebCore::ScheduledAction::executeFunctionInContext):
(WebCore::ScheduledAction::execute):
* Source/WebCore/bindings/js/ScriptCachedFrameData.cpp:
(WebCore::ScriptCachedFrameData::ScriptCachedFrameData):
(WebCore::ScriptCachedFrameData::restore):
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::~ScriptController):
(WebCore::ScriptController::evaluateInWorld):
(WebCore::ScriptController::linkAndEvaluateModuleScriptInWorld):
(WebCore::ScriptController::evaluateModule):
(WebCore::ScriptController::createWorld):
(WebCore::ScriptController::getAllWorlds):
(WebCore::ScriptController::initScriptForWindowProxy):
(WebCore::ScriptController::protectedFrame const):
(WebCore::ScriptController::setupModuleScriptHandlers):
(WebCore::ScriptController::windowProxy):
(WebCore::ScriptController::jsWindowProxy):
(WebCore::ScriptController::eventHandlerPosition const):
(WebCore::ScriptController::canAccessFromCurrentOrigin):
(WebCore::ScriptController::cacheableBindingRootObject):
(WebCore::ScriptController::bindingRootObject):
(WebCore::ScriptController::clearScriptObjects):
(WebCore::ScriptController::executeScriptIgnoringException):
(WebCore::ScriptController::executeScriptInWorld):
(WebCore::ScriptController::callInWorld):
(WebCore::ScriptController::canExecuteScripts):
(WebCore::ScriptController::executeJavaScriptURL):
(WebCore::ScriptController::reportExceptionFromScriptError):
(WebCore::ScriptController::registerImportMap):
(WebCore::ScriptController::isAcquiringImportMaps):
(WebCore::ScriptController::setAcquiringImportMaps):
(WebCore::ScriptController::setPendingImportMaps):
(WebCore::ScriptController::clearPendingImportMaps):
* Source/WebCore/bindings/js/ScriptController.h:
* Source/WebCore/bindings/js/ScriptControllerMac.mm:
(WebCore::ScriptController::disconnectPlatformScriptObjects):
* Source/WebCore/bindings/js/ScriptModuleLoader.cpp:
(WebCore::ScriptModuleLoader::resolve):
(WebCore::rejectToPropagateNetworkError):
(WebCore::rejectWithFetchError):
(WebCore::ScriptModuleLoader::fetch):
(WebCore::ScriptModuleLoader::responseURLFromRequestURL):
(WebCore::ScriptModuleLoader::evaluate):
(WebCore::ScriptModuleLoader::importModule):
(WebCore::ScriptModuleLoader::createImportMetaProperties):
(WebCore::ScriptModuleLoader::notifyFinished):
* Source/WebCore/bindings/js/StructuredClone.cpp:
(WebCore::JSC_DEFINE_HOST_FUNCTION):
* Source/WebCore/bindings/js/WebAssemblyCachedScriptSourceProvider.h:
* Source/WebCore/bindings/js/WebAssemblyScriptBufferSourceProvider.h:
* Source/WebCore/bindings/js/WebCoreJSClientData.cpp:
(WebCore::JSVMClientData::~JSVMClientData):
(WebCore::JSVMClientData::getAllWorlds):
(WebCore::JSVMClientData::overrideSourceURL const):
* Source/WebCore/bindings/js/WebCoreJSClientData.h:
(WebCore::JSVMClientData::rememberWorld):
(WebCore::JSVMClientData::forgetWorld):
* Source/WebCore/bindings/js/WindowProxy.cpp:
(WebCore::WindowProxy::protectedFrame const):
(WebCore::WindowProxy::detachFromFrame):
(WebCore::WindowProxy::replaceFrame):
(WebCore::WindowProxy::destroyJSWindowProxy):
(WebCore::WindowProxy::createJSWindowProxy):
(WebCore::WindowProxy::createJSWindowProxyWithInitializedScript):
(WebCore::WindowProxy::setDOMWindow):
* Source/WebCore/bindings/js/WindowProxy.h:
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::protectedWindow const):
* Source/WebCore/page/Frame.h:

Canonical link: <a href="https://commits.webkit.org/274647@main">https://commits.webkit.org/274647@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9d47becfae8f4f9765e4a143912eab3888a5fbf3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/39665 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/18644 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/42021 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/42200 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/35565 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/41971 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/21544 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15973 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/33107 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40239 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/15728 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/34325 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/13635 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/13626 "Found unexpected failure with change (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/35279 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/43477 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/33118 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/36018 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/35612 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/39397 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/39291 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/14477 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11925 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/37668 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/16083 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/46298 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/16132 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/9532 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5204 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/15740 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/bindings/js/JSDOMPromiseDeferred.cpp:
```diff
@@ -294,19 +294,21 @@ void fulfillPromiseWithArrayBuffer(Ref<DeferredPromise>&& promise, const void* d
 {
+    RefPtr globalScope = dynamicDowncast<WorkerGlobalScope>(*lexicalGlobalObject.scriptExecutionContext());
+    if (!globalScope)
+        return false;
```
The early return is much clearer!

brentfulgham reviewed (commented):
Will we be able to make similar changes to the generator scripts once the binding types are using smart pointers everywhere?

r=me

cdumez committed:
Adopt more smart pointers in WebCore/bindings
https://bugs.webkit.org/show_bug.cgi?id=269366

Reviewed by Brent Fulgham.

* Source/JavaScriptCore/runtime/JSObject.h:
* Source/WebCore/animation/WebAnimation.cpp:
(WebCore::WebAnimation::create):
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/bindings/js/CachedModuleScriptLoader.cpp:
(WebCore::CachedModuleScriptLoader::~CachedModuleScriptLoader):
(WebCore::CachedModuleScriptLoader::load):
(WebCore::CachedModuleScriptLoader::notifyFinished):
* Source/WebCore/bindings/js/CachedScriptSourceProvider.h:
(WebCore::CachedScriptSourceProvider::~CachedScriptSourceProvider):
(WebCore::CachedScriptSourceProvider::m_cachedScript):
(WebCore::CachedScriptSourceProvider::protectedScript const):
(WebCore::CachedScriptSourceProvider::hash const):
(WebCore::CachedScriptSourceProvider::source const):
* Source/WebCore/bindings/js/CommonVM.cpp:
(WebCore::commonVMSlow):
(WebCore::addImpureProperty):
(WebCore::protectedCommonVM):
* Source/WebCore/bindings/js/CommonVM.h:
* Source/WebCore/bindings/js/DOMGCOutputConstraint.h:
* Source/WebCore/bindings/js/DOMPromiseProxy.h:
(WebCore::DOMPromiseProxy<IDLType>::resolvePromise):
(WebCore::DOMPromiseProxy<IDLUndefined>::promise):
(WebCore::DOMPromiseProxyWithResolveCallback<IDLType>::promise):
* Source/WebCore/bindings/js/DOMWrapperWorld.cpp:
(WebCore::DOMWrapperWorld::~DOMWrapperWorld):
(WebCore::DOMWrapperWorld::clearWrappers):
(WebCore::DOMWrapperWorld::didCreateWindowProxy):
(WebCore::DOMWrapperWorld::didDestroyWindowProxy):
* Source/WebCore/bindings/js/DOMWrapperWorld.h:
(WebCore::DOMWrapperWorld::didCreateWindowProxy): Deleted.
(WebCore::DOMWrapperWorld::didDestroyWindowProxy): Deleted.
* Source/WebCore/bindings/js/GCController.cpp:
(WebCore::GCController::garbageCollectOnAlternateThreadForDebugging):
(WebCore::GCController::deleteAllCode):
(WebCore::GCController::deleteAllLinkedCode):
(WebCore::GCController::dumpHeap):
* Source/WebCore/bindings/js/IDBBindingUtilities.cpp:
(WebCore::get):
(WebCore::toJS):
(WebCore::createIDBKeyFromValue):
(WebCore::ensureNthValueOnKeyPath):
(WebCore::deserializeIDBValueToJSValue):
(WebCore::createKeyPathArray):
(WebCore::generateIndexKeyMapForValueIsolatedCopy):
(WebCore::deserializeIDBValueWithKeyInjection):
(WebCore::IDBSerializationContext::initializeVM):
* Source/WebCore/bindings/js/InternalReadableStream.cpp:
(WebCore::invokeReadableStreamFunction):
(WebCore::InternalReadableStream::isLocked const):
(WebCore::InternalReadableStream::isDisturbed const):
(WebCore::InternalReadableStream::cancel):
(WebCore::InternalReadableStream::lock):
(WebCore::InternalReadableStream::pipeTo):
(WebCore::InternalReadableStream::tee):
(WebCore::InternalReadableStream::getReader):
(WebCore::InternalReadableStream::pipeThrough):
* Source/WebCore/bindings/js/InternalWritableStream.cpp:
(WebCore::invokeWritableStreamFunction):
(WebCore::InternalWritableStream::createFromUnderlyingSink):
(WebCore::InternalWritableStream::locked const):
(WebCore::InternalWritableStream::lock):
(WebCore::InternalWritableStream::abortForBindings):
(WebCore::InternalWritableStream::closeForBindings):
(WebCore::InternalWritableStream::closeIfPossible):
(WebCore::InternalWritableStream::getWriter):
* Source/WebCore/bindings/js/JSCallbackData.cpp:
(WebCore::JSCallbackData::invokeCallback):
* Source/WebCore/bindings/js/JSCustomElementInterface.cpp:
(WebCore::JSCustomElementInterface::constructElementWithFallback):
(WebCore::JSCustomElementInterface::createElement):
(WebCore::JSCustomElementInterface::tryToConstructCustomElement):
(WebCore::constructCustomElementSynchronously):
(WebCore::JSCustomElementInterface::upgradeElement):
(WebCore::JSCustomElementInterface::invokeCallback):
(WebCore::JSCustomElementInterface::invokeFormStateRestoreCallback):
* Source/WebCore/bindings/js/JSCustomElementRegistryCustom.cpp:
(WebCore::getCustomElementCallback):
(WebCore::validateCustomElementNameAndThrowIfNeeded):
(WebCore::JSCustomElementRegistry::define):
(WebCore::whenDefinedPromise):
(WebCore::JSCustomElementRegistry::whenDefined):
* Source/WebCore/bindings/js/JSCustomEventCustom.cpp:
(WebCore::JSCustomEvent::detail const):
* Source/WebCore/bindings/js/JSDOMAbstractOperations.h:
(WebCore::isVisibleNamedProperty):
* Source/WebCore/bindings/js/JSDOMAsyncIterator.h:
(WebCore::IteratorTraits>::runNextSteps):
(WebCore::IteratorTraits>::getNextIterationResult):
(WebCore::IteratorTraits>::onPromiseSettled):
(WebCore::IteratorTraits>::createOnSettledFunction):
(WebCore::IteratorTraits>::onPromiseFulFilled):
(WebCore::IteratorTraits>::createOnFulfilledFunction):
(WebCore::IteratorTraits>::reject):
(WebCore::IteratorTraits>::onPromiseRejected):
(WebCore::IteratorTraits>::createOnRejectedFunction):
(WebCore::IteratorTraits>::next):
* Source/WebCore/bindings/js/JSDOMBindingSecurity.cpp:
(WebCore::printErrorMessageForFrame):
(WebCore::canAccessDocument):
(WebCore::BindingSecurity::shouldAllowAccessToFrame):
(WebCore::BindingSecurity::shouldAllowAccessToDOMWindow):
(WebCore::BindingSecurity::shouldAllowAccessToNode):
* Source/WebCore/bindings/js/JSDOMBindingSecurityInlines.h:
(WebCore::BindingSecurity::shouldAllowAccessToDOMWindow):
* Source/WebCore/bindings/js/JSDOMBuiltinConstructor.h:
(WebCore::JSDOMBuiltinConstructor<JSClass>::getDOMStructureForJSObject):
* Source/WebCore/bindings/js/JSDOMCastThisValue.h:
(WebCore::castThisValue):
* Source/WebCore/bindings/js/JSDOMConstructorBase.cpp:
(WebCore::JSC_DEFINE_HOST_FUNCTION):
* Source/WebCore/bindings/js/JSDOMConvertAny.h:
(WebCore::VariadicConverter<IDLAny>::convert):
* Source/WebCore/bindings/js/JSDOMConvertBufferSource.h:
(WebCore::Detail::BufferSourceConverter::convert):
* Source/WebCore/bindings/js/JSDOMConvertCallbacks.h:
(WebCore::Converter<IDLCallbackFunction<T>>::convert):
(WebCore::Converter<IDLCallbackInterface<T>>::convert):
* Source/WebCore/bindings/js/JSDOMConvertDate.cpp:
(WebCore::valueToDate):
* Source/WebCore/bindings/js/JSDOMConvertEnumeration.h:
(WebCore::Converter<IDLEnumeration<T>>::convert):
* Source/WebCore/bindings/js/JSDOMConvertEventListener.h:
(WebCore::Converter<IDLEventListener<T>>::convert):
* Source/WebCore/bindings/js/JSDOMExceptionHandling.cpp:
(WebCore::reportException):
(WebCore::reportCurrentException):
(WebCore::createDOMException):
* Source/WebCore/bindings/js/JSDOMGlobalObject.cpp:
(WebCore::JSC_DEFINE_HOST_FUNCTION):
(WebCore::JSDOMGlobalObject::protectedScriptExecutionContext const):
(WebCore::JSDOMGlobalObject::promiseRejectionTracker):
(WebCore::JSDOMGlobalObject::createCrossOriginFunction):
(WebCore::JSDOMGlobalObject::createCrossOriginGetterSetter):
(WebCore::handleResponseOnStreamingAction):
(WebCore::JSDOMGlobalObject::moduleLoaderFetch):
(WebCore::JSDOMGlobalObject::moduleLoaderImportModule):
(WebCore::JSDOMGlobalObject::deriveShadowRealmGlobalObject):
(WebCore::JSDOMGlobalObject::protectedWorld):
(WebCore::toJSDOMGlobalObject):
(WebCore::callerGlobalObject):
* Source/WebCore/bindings/js/JSDOMGlobalObject.h:
* Source/WebCore/bindings/js/JSDOMGuardedObject.cpp:
(WebCore::DOMGuardedObject::DOMGuardedObject):
* Source/WebCore/bindings/js/JSDOMIterator.cpp:
(WebCore::addValueIterableMethods):
* Source/WebCore/bindings/js/JSDOMIterator.h:
(WebCore::iteratorForEach):
(WebCore::IteratorTraits>::next):
* Source/WebCore/bindings/js/JSDOMMapLike.h:
(WebCore::forwardSizeToMapLike):
(WebCore::forwardEntriesToMapLike):
(WebCore::forwardKeysToMapLike):
(WebCore::forwardValuesToMapLike):
(WebCore::forwardClearToMapLike):
(WebCore::forwardGetToMapLike):
(WebCore::forwardHasToMapLike):
(WebCore::forwardSetToMapLike):
(WebCore::forwardDeleteToMapLike):
* Source/WebCore/bindings/js/JSDOMMicrotask.cpp:
(WebCore::JSDOMMicrotask::run):
* Source/WebCore/bindings/js/JSDOMOperation.h:
(WebCore::IDLOperation::call):
* Source/WebCore/bindings/js/JSDOMPromise.cpp:
(WebCore::DOMPromise::whenPromiseIsSettled):
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.cpp:
(WebCore::DeferredPromise::callFunction):
(WebCore::DeferredPromise::whenSettled):
(WebCore::DeferredPromise::reject):
(WebCore::createRejectedPromiseWithTypeError):
(WebCore::DeferredPromise::handleTerminationExceptionIfNeeded):
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.h:
(WebCore::DeferredPromise::resolve):
(WebCore::DeferredPromise::resolveWithNewlyCreated):
(WebCore::DeferredPromise::resolveCallbackValueWithNewlyCreated):
(WebCore::DeferredPromise::reject):
(WebCore::DeferredPromise::resolveWithCallback):
(WebCore::DeferredPromise::rejectWithCallback):
(WebCore::DOMPromiseDeferredBase::reject):
(WebCore::DOMPromiseDeferredBase::rejectType):
(WebCore::DOMPromiseDeferredBase::whenSettled):
(WebCore::DOMPromiseDeferredBase::protectedPromise const):
(WebCore::DOMPromiseDeferred::resolve):
(WebCore::DOMPromiseDeferred<void>::resolve):
(WebCore::callPromiseFunction):
(WebCore::callPromisePairFunction):
* Source/WebCore/bindings/js/JSDOMSetLike.cpp:
(WebCore::getBackingSet):
(WebCore::clearBackingSet):
(WebCore::addToBackingSet):
(WebCore::forwardFunctionCallToBackingSet):
* Source/WebCore/bindings/js/JSDOMSetLike.h:
(WebCore::getAndInitializeBackingSet):
(WebCore::forwardSizeToSetLike):
(WebCore::forwardEntriesToSetLike):
(WebCore::forwardKeysToSetLike):
(WebCore::forwardValuesToSetLike):
(WebCore::forwardClearToSetLike):
(WebCore::forwardHasToSetLike):
(WebCore::forwardAddToSetLike):
(WebCore::forwardDeleteToSetLike):
* Source/WebCore/bindings/js/JSDOMWindowBase.cpp:
(WebCore::JSDOMWindowBase::updateDocument):
(WebCore::JSDOMWindowBase::printErrorMessage const):
(WebCore::JSDOMWindowBase::supportsRichSourceInfo):
(WebCore::JSDOMWindowBase::shouldInterruptScript):
(WebCore::JSDOMWindowBase::shouldInterruptScriptBeforeTimeout):
(WebCore::JSDOMWindowBase::javaScriptRuntimeFlags):
(WebCore::JSDOMWindowBase::queueMicrotaskToEventLoop):
(WebCore::JSDOMWindowBase::currentScriptExecutionOwner):
(WebCore::JSDOMWindowBase::fireFrameClearedWatchpointsForWindow):
* Source/WebCore/bindings/js/JSDOMWindowProperties.cpp:
(WebCore::jsDOMWindowPropertiesGetOwnPropertySlotNamedItemGetter):
(WebCore::JSDOMWindowProperties::finishCreation):
(WebCore::JSDOMWindowProperties::getOwnPropertySlot):
(WebCore::JSDOMWindowProperties::getOwnPropertySlotByIndex):
(WebCore::JSDOMWindowProperties::defineOwnProperty):
* Source/WebCore/bindings/js/JSDOMWrapper.cpp:
(WebCore::cloneAcrossWorlds):
* Source/WebCore/bindings/js/JSDOMWrapperCache.h:
(WebCore::setSubclassStructureIfNeeded):
* Source/WebCore/bindings/js/JSDocumentCustom.cpp:
(WebCore::cachedDocumentWrapper):
(WebCore::reportMemoryForDocumentIfFrameless):
(WebCore::setAdoptedStyleSheetsOnTreeScope):
(WebCore::JSDocument::setAdoptedStyleSheets):
* Source/WebCore/bindings/js/JSElementCustom.cpp:
(WebCore::toJS):
(WebCore::toJSNewlyCreated):
(WebCore::getElementsArrayAttribute):
* Source/WebCore/bindings/js/JSElementInternalsCustom.cpp:
(WebCore::JSElementInternals::setFormValue):
(WebCore::getElementsArrayAttribute):
* Source/WebCore/bindings/js/JSErrorHandler.cpp:
(WebCore::JSErrorHandler::handleEvent):
* Source/WebCore/bindings/js/JSEventListener.cpp:
(WebCore::JSEventListener::handleEvent):
(WebCore::JSEventListener::functionName const):
* Source/WebCore/bindings/js/JSEventListener.h:
(WebCore::windowEventHandlerAttribute):
(WebCore::setWindowEventHandlerAttribute):
(WebCore::JSEventListener::ensureJSFunction const):
* Source/WebCore/bindings/js/JSEventTargetCustom.h:
(WebCore::JSEventTargetWrapper::JSEventTargetWrapper):
(WebCore::IDLOperation<JSEventTarget>::call):
* Source/WebCore/bindings/js/JSExecState.cpp:
(WebCore::JSExecState::didLeaveScriptContext):
* Source/WebCore/bindings/js/JSExecState.h:
(WebCore::JSExecState::call):
(WebCore::JSExecState::evaluate):
(WebCore::JSExecState::profiledCall):
(WebCore::JSExecState::profiledEvaluate):
(WebCore::JSExecState::linkAndEvaluateModule):
(WebCore::JSExecState::~JSExecState):
* Source/WebCore/bindings/js/JSExtendableMessageEventCustom.cpp:
(WebCore::constructJSExtendableMessageEvent):
(WebCore::JSExtendableMessageEvent::data const):
* Source/WebCore/bindings/js/JSHTMLAllCollectionCustom.cpp:
(WebCore::JSC_DEFINE_HOST_FUNCTION):
* Source/WebCore/bindings/js/JSHTMLElementCustom.cpp:
(WebCore::constructJSHTMLElement):
(WebCore::JSHTMLElement::pushEventHandlerScope const):
* Source/WebCore/bindings/js/JSHistoryCustom.cpp:
(WebCore::JSHistory::state const):
* Source/WebCore/bindings/js/JSImageDataCustom.cpp:
(WebCore::toJSNewlyCreated):
* Source/WebCore/bindings/js/JSKeyframeEffectCustom.cpp:
(WebCore::JSKeyframeEffect::getKeyframes):
* Source/WebCore/bindings/js/JSLazyEventListener.cpp:
(WebCore::JSLazyEventListener::JSLazyEventListener):
(WebCore::JSLazyEventListener::initializeJSFunction const):
(WebCore::JSLazyEventListener::create):
* Source/WebCore/bindings/js/JSLocalDOMWindowCustom.cpp:
(WebCore::JSC_DEFINE_CUSTOM_GETTER):
(WebCore::jsLocalDOMWindowGetOwnPropertySlotRestrictedAccess):
(WebCore::JSLocalDOMWindow::getOwnPropertySlot):
(WebCore::JSLocalDOMWindow::getOwnPropertySlotByIndex):
(WebCore::JSLocalDOMWindow::put):
(WebCore::JSLocalDOMWindow::putByIndex):
(WebCore::JSLocalDOMWindow::deleteProperty):
(WebCore::JSLocalDOMWindow::deletePropertyByIndex):
(WebCore::addCrossOriginOwnPropertyNames):
(WebCore::addScopedChildrenIndexes):
(WebCore::JSLocalDOMWindow::getOwnPropertyNames):
(WebCore::JSLocalDOMWindow::defineOwnProperty):
(WebCore::JSLocalDOMWindow::getPrototype):
(WebCore::DialogHandler::dialogCreated):
(WebCore::DialogHandler::returnValue const):
(WebCore::JSC_DEFINE_HOST_FUNCTION):
(WebCore::JSLocalDOMWindow::queueMicrotask):
(WebCore::JSLocalDOMWindow::setOpener):
(WebCore::jsLocalDOMWindowInstanceFunction_openDatabaseBody):
(WebCore::JSLocalDOMWindow::openDatabase const):
(WebCore::JSLocalDOMWindow::setOpenDatabase):
* Source/WebCore/bindings/js/JSLocationCustom.cpp:
(WebCore::getOwnPropertySlotCommon):
(WebCore::JSLocation::getOwnPropertySlot):
(WebCore::JSLocation::getOwnPropertySlotByIndex):
(WebCore::JSLocation::put):
(WebCore::JSLocation::putByIndex):
(WebCore::JSLocation::deleteProperty):
(WebCore::JSLocation::deletePropertyByIndex):
(WebCore::JSLocation::getOwnPropertyNames):
(WebCore::JSLocation::defineOwnProperty):
(WebCore::JSLocation::getPrototype):
* Source/WebCore/bindings/js/JSMicrotaskCallback.h:
(WebCore::JSMicrotaskCallback::call):
* Source/WebCore/bindings/js/JSNodeCustom.cpp:
(WebCore::willCreatePossiblyOrphanedTreeByRemovalSlowCase):
* Source/WebCore/bindings/js/JSObservableArray.cpp:
(JSC::JSC_DEFINE_CUSTOM_GETTER):
(JSC::JSObservableArray::getOwnPropertyNames):
(JSC::JSObservableArray::getOwnPropertySlot):
(JSC::JSObservableArray::put):
(JSC::JSObservableArray::deleteProperty):
(JSC::JSObservableArray::defineOwnProperty):
* Source/WebCore/bindings/js/JSObservableArray.h:
* Source/WebCore/bindings/js/JSPopStateEventCustom.cpp:
(WebCore::JSPopStateEvent::state const):
* Source/WebCore/bindings/js/JSRTCRtpSFrameTransformCustom.cpp:
(WebCore::JSRTCRtpSFrameTransform::setEncryptionKey):
* Source/WebCore/bindings/js/JSReadableStreamSourceCustom.cpp:
(WebCore::JSReadableStreamSource::start):
(WebCore::JSReadableStreamSource::pull):
* Source/WebCore/bindings/js/JSRemoteDOMWindowBase.cpp:
(WebCore::JSRemoteDOMWindowBase::protectedWrapped const):
* Source/WebCore/bindings/js/JSRemoteDOMWindowBase.h:
* Source/WebCore/bindings/js/JSRemoteDOMWindowCustom.cpp:
(WebCore::JSRemoteDOMWindow::getOwnPropertySlotByIndex):
(WebCore::JSRemoteDOMWindow::put):
(WebCore::JSRemoteDOMWindow::deleteProperty):
(WebCore::JSRemoteDOMWindow::deletePropertyByIndex):
(WebCore::JSRemoteDOMWindow::defineOwnProperty):
* Source/WebCore/bindings/js/JSShadowRootCustom.cpp:
(WebCore::JSShadowRoot::setAdoptedStyleSheets):
* Source/WebCore/bindings/js/JSWebAnimationCustom.cpp:
(WebCore::constructJSWebAnimation):
* Source/WebCore/bindings/js/JSWindowProxy.cpp:
(WebCore::JSWindowProxy::setWindow):
* Source/WebCore/bindings/js/JSWorkerGlobalScopeBase.cpp:
(WebCore::JSWorkerGlobalScopeBase::queueMicrotaskToEventLoop):
* Source/WebCore/bindings/js/JSWorkerGlobalScopeCustom.cpp:
(WebCore::JSWorkerGlobalScope::queueMicrotask):
* Source/WebCore/bindings/js/ModuleScriptLoader.h:
(WebCore::ModuleScriptLoader::ModuleScriptLoader):
* Source/WebCore/bindings/js/ModuleScriptLoaderClient.h:
* Source/WebCore/bindings/js/ReadableStreamDefaultController.cpp:
(WebCore::invokeReadableStreamDefaultControllerFunction):
(WebCore::ReadableStreamDefaultController::error):
(WebCore::ReadableStreamDefaultController::enqueue):
* Source/WebCore/bindings/js/ScheduledAction.cpp:
(WebCore::ScheduledAction::executeFunctionInContext):
(WebCore::ScheduledAction::execute):
* Source/WebCore/bindings/js/ScriptCachedFrameData.cpp:
(WebCore::ScriptCachedFrameData::ScriptCachedFrameData):
(WebCore::ScriptCachedFrameData::restore):
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::~ScriptController):
(WebCore::ScriptController::evaluateInWorld):
(WebCore::ScriptController::linkAndEvaluateModuleScriptInWorld):
(WebCore::ScriptController::evaluateModule):
(WebCore::ScriptController::createWorld):
(WebCore::ScriptController::getAllWorlds):
(WebCore::ScriptController::initScriptForWindowProxy):
(WebCore::ScriptController::protectedFrame const):
(WebCore::ScriptController::setupModuleScriptHandlers):
(WebCore::ScriptController::windowProxy):
(WebCore::ScriptController::jsWindowProxy):
(WebCore::ScriptController::eventHandlerPosition const):
(WebCore::ScriptController::canAccessFromCurrentOrigin):
(WebCore::ScriptController::cacheableBindingRootObject):
(WebCore::ScriptController::bindingRootObject):
(WebCore::ScriptController::clearScriptObjects):
(WebCore::ScriptController::executeScriptIgnoringException):
(WebCore::ScriptController::executeScriptInWorld):
(WebCore::ScriptController::callInWorld):
(WebCore::ScriptController::canExecuteScripts):
(WebCore::ScriptController::executeJavaScriptURL):
(WebCore::ScriptController::reportExceptionFromScriptError):
(WebCore::ScriptController::registerImportMap):
(WebCore::ScriptController::isAcquiringImportMaps):
(WebCore::ScriptController::setAcquiringImportMaps):
(WebCore::ScriptController::setPendingImportMaps):
(WebCore::ScriptController::clearPendingImportMaps):
* Source/WebCore/bindings/js/ScriptController.h:
* Source/WebCore/bindings/js/ScriptControllerMac.mm:
(WebCore::ScriptController::disconnectPlatformScriptObjects):
* Source/WebCore/bindings/js/ScriptModuleLoader.cpp:
(WebCore::ScriptModuleLoader::resolve):
(WebCore::rejectToPropagateNetworkError):
(WebCore::rejectWithFetchError):
(WebCore::ScriptModuleLoader::fetch):
(WebCore::ScriptModuleLoader::responseURLFromRequestURL):
(WebCore::ScriptModuleLoader::evaluate):
(WebCore::ScriptModuleLoader::importModule):
(WebCore::ScriptModuleLoader::createImportMetaProperties):
(WebCore::ScriptModuleLoader::notifyFinished):
* Source/WebCore/bindings/js/StructuredClone.cpp:
(WebCore::JSC_DEFINE_HOST_FUNCTION):
* Source/WebCore/bindings/js/WebAssemblyCachedScriptSourceProvider.h:
* Source/WebCore/bindings/js/WebAssemblyScriptBufferSourceProvider.h:
* Source/WebCore/bindings/js/WebCoreJSClientData.cpp:
(WebCore::JSVMClientData::~JSVMClientData):
(WebCore::JSVMClientData::getAllWorlds):
(WebCore::JSVMClientData::overrideSourceURL const):
* Source/WebCore/bindings/js/WebCoreJSClientData.h:
(WebCore::JSVMClientData::rememberWorld):
(WebCore::JSVMClientData::forgetWorld):
* Source/WebCore/bindings/js/WindowProxy.cpp:
(WebCore::WindowProxy::protectedFrame const):
(WebCore::WindowProxy::detachFromFrame):
(WebCore::WindowProxy::replaceFrame):
(WebCore::WindowProxy::destroyJSWindowProxy):
(WebCore::WindowProxy::createJSWindowProxy):
(WebCore::WindowProxy::createJSWindowProxyWithInitializedScript):
(WebCore::WindowProxy::setDOMWindow):
* Source/WebCore/bindings/js/WindowProxy.h:
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::protectedWindow const):
* Source/WebCore/page/Frame.h:

Canonical link: https://commits.webkit.org/274647@main
